### PR TITLE
Add Size parameter to PtrStepSz<T>

### PR DESCRIPTION
**Pull request by srinath2468**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by opencv-alalek** [comment]
Need to completely remove commits with submodules from the PR.

**Comment by srinath2468** [comment]
@opencv-alalek  I have removed the commits with submodules

**Comment by asmorkalov** [comment]
What issue do you try to solve? The PR does not contain enough add value to be merged.

**Comment by srinath2468** [comment]
@asmorkalov The PR does not solve any issue. It is just an added feature. PtrStepSz seems to not give us access to size but rather cols and rows. To check for size, we need to construct using these parameters. It would be helpful if the type gave us access to size directly to use in custom functions.

**Comment by pasbi** [comment]
@asmorkalov I vote to open this PR again.

Having a handy way to access the size rather than just rows and cols would make our code more terse and avoids the pitfall of confusing rows and cols (you know, OpenCV is a bit... peculiar in this regard).

However, I'd add `size` as a function which assembles a `cv::Size` from `rows` and `cols` just like `cv::Mat` does, except that I'd add decorators to make it accessible from kernels and device functions.

**Comment by srinath2468** [comment]
@asmorkalov thanks for reopening. Do you think the current proposal makes sense? Or should we construct `cv::Size` in gpu code with a getter method. This will keep the overhead of `PtrStepSz..` the same.

**Comment by asmorkalov** [comment]
The PR was discussed on the Core team meeting and decided to reject it. The PR makes the structure 2 times fatter and values duplication. Closed.

File: modules/core/include/opencv2/core/cuda_types.hpp [hunk 1]
```diff
@@ -120,10 +120,11 @@ namespace cv
                 : PtrStep<T>(data_, step_), cols(cols_), rows(rows_) {}
 
             template <typename U>
-            explicit PtrStepSz(const PtrStepSz<U>& d) : PtrStep<T>((T*)d.data, d.step), cols(d.cols), rows(d.rows){}
+            explicit PtrStepSz(const PtrStepSz<U>& d) : PtrStep<T>((T*)d.data, d.step), cols(d.cols), rows(d.rows), size(d.cols,d.rows) {}
```
Comment (pasbi) [inline modules/core/include/opencv2/core/cuda_types.hpp:123]:
```
explicit PtrStepSz(const PtrStepSz<U>& d) : PtrStep<T>((T*)d.data, d.step), cols(d.cols), rows(d.rows) {}
```

**Comment by pasbi** [comment]
@asmorkalov I totally understand your decision.

However, what about my proposal: having a function `size()` instead? It wouldn't increase the size of the struct and avoid value duplication.

@srinath2468 maybe you could update your pull request?

Please consider to merge this PR with my proposed changed.

**Comment by asmorkalov** [comment]
size() makes sense then, if it's used.

**Pull request closed without merging**